_posts/2016-04-04-code-smells-part-II.md

```diff
@@ -0,0 +1,88 @@
+---
+layout: post
+name: code-smells-part-one
+title: Code Smells – Part II
+date: 2016-04-04 12:10:00 +00:00
+author: Ana Nogal
+canonical:
+    name: my personal blog
+    href: http://www.ananogal.com/blog/code-smells-part-two/
+image:
+    src: /assets/img/custom/blog/code-smells_II.jpg
+    attribution:
+        text: Via Flickr/Creative Commons/Brian Fitzgerald (Creative Commons - Attribution-ShareAlike 2.0 Generic license)
+        href: https://www.flickr.com/photos/brian-fitzgerald/3334353375
+tags:
+- object-orientation abusers
+- change preventers
+- code smells
+- refactoring
+
+---
+In the last post, [Code Smells - Part I](http://codurance.com/2016/03/17/code-smells-part-I),  I talked about the bloaters: they are code smells that can be identified as Long Methods, Large Classes, Primitive Obsessions, Long Parameter List and Data Clumps. In this one, I would like to dig into the **Object-Orientation Abusers** and the **Change Preventers**. 
+In today's post, I'll talk about a few refactoring techniques, but there are a lot more. You can find a good reference to all of then in [Refactoring.com](http://refactoring.com/catalog/) by Martin Fowler.
+
+## Object-Orientation Abusers
+
+This type of code smell usually happens when the object-oriented principles are incomplete or incorrectly applied.
+
+
+#### Switch Statements
+
+This case is simple to identify: we have a switch case. But you should consider a smell too if you find a sequence of ifs. (that's a switch case in disguise).
+Why switch statements are bad? Because when a new condition is added, you have to find all the switch code and modify it. 
+As a rule of thumb, when you see a switch statement you should think of polymorphism. There are two refactoring techniques that we can apply here: 
+
+ - **_Replace Type Code with Subclasses_**
+   This technique consists of creating subclasses for each type of your switch, putting the behaviour in this classes. 
+ - **_Replace Type Code With Strategy_**
+   Similar to the above one, in this case, you should make use of one of the patterns: [State](https://en.wikipedia.org/wiki/State_pattern) or [Strategy](https://en.wikipedia.org/wiki/Strategy_pattern).
+
+
+#### Temporary Field
+
+This case occurs when we are calculating some big algorithm that needs several amount of inputs. Creating this fields in the class makes then have no value most of the times because they are just used for this specific calculation. Here the best refactoring technique is to use **_Replace Method with Method Object_**, wich will extract the method into a separate class. Then you can split the method into several methods within the same class. 
```

Comment:
I think those fields are also dangerous because your need to make sure to reinitialize them for the next computation